keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,318 @@
+
+#### Address scalability issues with the disruption controller
+
+The disruption controller has some performance issues as reported in https://github.com/kubernetes/kubernetes/issues/92826.
+This should be addressed before GA.
+
+#### Fix handling of empty selector in disruption controller
+
+The disruption controller doesn't handle empty selector correctly https://github.com/kubernetes/kubernetes/issues/95083. 
+We can't fix this directly in the v1beta1 version of PDBs, but we should fix this as part of promoting PDBs to v1.
+
+### API changes
+
+* Add conditions to the status object for PodDisruptionBudget.
+* Change the semantics of an empty selector.
+
+
+### Test Plan
+
+#### Existing Tests
+PodDisruptionBudget currently has tests in various components that use the feature:
+
+* Unit tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/disruption/disruption_test.go
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/disruption/disruption_test.go
+* Kubectl
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate/versioned/pdb_test.go
+  tests generation of a PDB objects out of given parameters
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create/create_pdb_test.go
+  tests creation of PDB objects from cmd parameters
+* Scheduler
+  - https://github.com/kubernetes/kubernetes/blob/4b59044b8d2a3502ea490ba2c958008a098511a3/test/integration/scheduler/preemption_test.go#L1053
+  tests effects of PDB on preemption (PDB is honored in a best effort way)
+* Eviction integration tests
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/evictions/evictions_test.go test eviction logic and its interactions with PDB.
+* Autoscaler
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/cluster_size_autoscaling.go ensure that Autoscaler respects PDB when draining nodes.
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/disruption.go
+
+All beta PDB endpoints are covered by e2e tests: https://apisnoop.cncf.io/1.20.0/beta/policy
+
+#### Conformance tests
+
+The following e2e tests will be included in the conformance tests:
+
+ * DisruptionController
+   * should block an eviction until the PDB is updated to allow it
+   * create a PodDisruptionBudget
+   * delete collection of PodDisruptionBudget
+   * delete a PodDisruptionBudget
+   * list or watch objects of kind PodDisruptionBudget
+   * partially update the specified PodDisruptionBudget
+   * partially update status of the specified PodDisruptionBudget
+   * read the specified PodDisruptionBudget
+   * read status of the specified PodDisruptionBudget
+   * replace the specified PodDisruptionBudget
+   * replace status of the specified PodDisruptionBudget
+
+### Graduation Criteria
+
```

Comment:
I'm assuming this Graduation Criteria is for moving to GA, if you could make that explicit here that would be great :smile:

Reply:
Updated the language to make this clear.